source/Missions/Services/Agents/Messages/AgentPublisher.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Common.Logging;
+using Common.Messaging;
+using Missions.Services.Network;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Messages
```

Comment:
This should exist in `Missions.Services.Agents`, or one level up in the directory